### PR TITLE
Generate .report-build.json

### PR DESCRIPTION
**Pull request by live1206**
[AB#237845](https://dev.azure.com/ceapex/Engineering/_workitems/edit/237845)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6101)

### CODE REVIEW

File: src/docfx/build/Build.cs [hunk 1]
```diff
@@ -109,6 +110,7 @@ private static void Run(Context context)
                 () => context.Output.WriteJson(".publish.json", publishModel),
                 () => context.Output.WriteJson(".dependencymap.json", dependencyMap.ToDependencyMapModel()),
                 () => context.Output.WriteJson(".links.json", context.FileLinkMapBuilder.Build(context.PublishUrlMap.GetAllFiles())),
+                () => context.Output.WriteJson(".report-build.json", context.ReportModelBuilder.Build()),
```
Comment (yufeih) [inline src/docfx/build/Build.cs:113]: `.buildreport.json`?

File: src/docfx/config/ops/OpsConfigAdapter.cs [hunk 2]
```diff
@@ -250,9 +252,9 @@ private async Task<string> FetchValidationRules(string url, IReadOnlyDictionary<
                                }
                            }
                            var response = await _http.SendAsync(request);
-                           if (response.Headers.TryGetValues("X-Metadata-Version", out var metadataVersion))
+                           if (_reportModelBuilder != null && response.Headers.TryGetValues("X-Metadata-Version", out var metadataVersion))
                            {
-                               _errorLog.Write(Errors.System.MetadataValidationRuleset(string.Join(',', metadataVersion)));
```
Comment (yufeih) [inline src/docfx/config/ops/OpsConfigAdapter.cs:255]: keep this line before the feature is done

**Comment by live1206** [comment]
Postpone this work until we have a clear contract for reporting

**Pull request closed without merging**